### PR TITLE
[url_launcher] Switch to new launchUrl interface

### PR DESCRIPTION
**Pull request by stuartmorgan**
Switches the new `launchUrl*` API methods to use the new corresponding
platform interface method instead of the legacy version.

The legacy `launch` API still uses the old interface to ensure that its
behavior doesn't change.

Follow-up to https://github.com/flutter/plugins/pull/5966

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/main/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Review by ditman — COMMENTED** [review]
There's a couple of test failing (in the web, it seems), @stuartmorgan do you want me to take a look and land this, or are you still working on it?

**Comment by stuartmorgan** [comment]
It's on my list, I just haven't had a chance to get back to it. Hopefully tomorrow.

**Comment by stuartmorgan** [comment]
Oops, I missed that you hadn't approved. Ready for another review!

File: packages/url_launcher/url_launcher/lib/src/url_launcher_string.dart [hunk 1]
```diff
@@ -2,6 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'package:url_launcher/src/type_conversion.dart';
```
Comment (bparrishMines) [inline packages/url_launcher/url_launcher/lib/src/url_launcher_string.dart:5]: nit: relative url

Comment (stuartmorgan) [reply packages/url_launcher/url_launcher/lib/src/url_launcher_string.dart:5]: Good catch, I noticed and fixed the other file's copy, but missed this one. I wish VS Code weren't so eager to add package-based imports.

**Pull request merged**